streaming_main.py

```diff
@@ -0,0 +1,596 @@
+            )
+
+            def update_progress(value):
+                progress_mutex.acquire()
+                try:
+                    progressbar.update(value)
+                finally:
+                    progress_mutex.release()
+
+            self.update_function = update_progress
+
+        start = datetime.datetime.now()
+
+        with concurrent.futures.ThreadPoolExecutor(max_workers=workers) as executor:
+            futures = [
+                executor.submit(self.process_list, file_list)
+                for file_list in work_lists
+            ]
+
+            for future in concurrent.futures.as_completed(futures):
+                test_runs = future.result()
+                for test_run in test_runs:
+                    self.count_result(test_run)
+                    if self.verbose or (
+                        self.fail_only and test_run.result not in NON_FAIL_RESULTS
+                    ):
+                        print(
+                            f"{EMOJIS[test_run.result]} {test_run.file}"
+                            f"{' (strict mode)' if test_run.strict_mode else ''}"
+                        )
+                        if test_run.output:
+                            print()
+                            print(test_run.output)
+                            print()
+
+                        if test_run.exit_code:
+                            signalnum = test_run.exit_code * -1
+                            if not test_run.output:
+                                print()
+                                print(f"{signal.strsignal(signalnum)}: {signalnum}")
+                                print()
+
+        if not self.silent:
+            progressbar.close()
+
+        end = datetime.datetime.now()
+        self.duration = end - start
+
+        self.log(f"Finished running tests in {self.duration}.")
+
+
+def main() -> None:
+    parser = ArgumentParser(
+        description="Run the test262 ECMAScript test suite with SerenityOS's LibJS",
+        epilog=", ".join(f"{EMOJIS[result]} = {result.value}" for result in TestResult),
+    )
+    parser.add_argument(
+        "-j",
+        "--libjs-test262-runner",
+        required=True,
+        metavar="PATH",
+        help="path to the 'libjs-test262-runner' binary",
+    )
+    parser.add_argument(
+        "-b",
+        "--use-bytecode",
+        action="store_true",
+        help="Use the bytecode interpreter to run the tests",
+    )
+    parser.add_argument(
+        "-t",
+        "--test262-root",
+        required=True,
+        metavar="PATH",
+        help="path to the 'test262' directory",
+    )
+    parser.add_argument(
+        "-p",
+        "--pattern",
+        default="test/**/*.js",
+        help="glob pattern used for test file searching (defaults to test/**/*.js)",
+    )
+    parser.add_argument(
+        "-c",
+        "--concurrency",
+        default=CPU_COUNT // 2,
```

Comment:
The default is now `CPU / 2` since it pushes much harder.
Whereas before I usually just ran it in the background with `CPU` workers it uses the full CPU.
We can change this but in my testing it does not matter that much probably due to hyperthreading and not having 16 real cores.

Reply:
If we do include this change the runner workflow file will have to override it, since that's the only thing running there and so its useless to not use all of it's cores.

Reply:
Yes you can still set it with `-c 32` or however many you need.
Or maybe we can use `-1` to set it to `CPU_COUNT`? Meaning we don't have to hardcode anything.

Also I currently have no PR in the work to make the runner use this as I would like a little more tests on different machines to ensure it is stable and correct.

Reply:
I added the `-1` means full CPU option for now.